src/BitStringAddresses/fermifs.jl

```diff
@@ -0,0 +1,213 @@
+    S = typeof(BitString{M}(0))
+    return FermiFS{N,M,S}(SVector{M}(onr))
+end
+function FermiFS{N}(onr::Union{SVector{M},Tuple{M}}) where {N,M}
+    return FermiFS{N,M}(onr)
+end
+function FermiFS(onr::Union{AbstractVector,Tuple})
+    M = length(onr)
+    N = sum(onr)
+    return FermiFS{N,M}(onr)
+end
+
+function Base.show(io::IO, f::FermiFS{N,M}) where {N,M}
+    print(io, "FermiFS{$N,$M}(", tuple(onr(f)...), ")")
+end
+
+Base.isless(a::FermiFS, b::FermiFS) = isless(a.bs, b.bs)
+Base.hash(a::FermiFS,  h::UInt) = hash(a.bs, h)
+Base.:(==)(a::FermiFS, b::FermiFS) = a.bs == b.bs
+num_particles(::Type{FermiFS{N,M,S}}) where {N,M,S} = N
+num_modes(::Type{FermiFS{N,M,S}}) where {N,M,S} = M
+num_components(::Type{<:FermiFS}) = 1
+
+function near_uniform(::Type{FermiFS{N,M}}) where {N,M}
+    return FermiFS([fill(1, N); fill(0, M - N)])
+end
+
+onr(a::FermiFS) = SVector(m_onr(a))
+
+struct FermiOccupiedOrbitals{N,S}
+    bs::S
+end
+
+"""
+    occupied_orbitals(f::FermiFS)
+
+Iterate over occupied orbitals in `FermiFS` address. Iterates values of type `Int`.
+
+# Example
+
+```jldoctest
+julia> f = FermiFS((1,1,0,1,0,0,1))
+julia> for i in occupied_orbitals(f)
+    @show i
+end
+i = 1
+i = 2
+i = 4
+i = 7
+
+"""
+occupied_orbitals(a::FermiFS{N,<:Any,S}) where {N,S} = FermiOccupiedOrbitals{N,S}(a.bs)
+
+function Base.iterate(o::FermiOccupiedOrbitals)
+    c = 0
+    chunk = o.bs.chunks[end]
+    while iszero(chunk)
+        c += 1
+        chunk = o.bs.chunks[end - c]
+    end
+    zeros = trailing_zeros(chunk % Int)
+    return iterate(o, (chunk >> (zeros % UInt64), c * 64 + zeros, c))
+end
+function Base.iterate(o::FermiOccupiedOrbitals, st)
+    chunk, index, c = st
+    while iszero(chunk)
+        c += 1
+        c == num_chunks(o.bs) && return nothing
+        chunk = o.bs.chunks[end - c]
+        index = c * 64
+    end
+    zeros = trailing_zeros(chunk % Int)
+    index += zeros
+    chunk >>= zeros
+    return index + 1, (chunk >> 1, index + 1, c)
+end
+
+function Base.iterate(o::FermiOccupiedOrbitals{<:Any,<:BitString{<:Any,1,T}}) where {T}
+    chunk = o.bs.chunks[end]
+    zeros = trailing_zeros(chunk % Int)
+    return iterate(o, (chunk >> (zeros % T), zeros))
+end
+function Base.iterate(o::FermiOccupiedOrbitals{<:Any,<:BitString{<:Any,1,T}}, st) where {T}
+    chunk, index = st
+    iszero(chunk) && return nothing
+    chunk >>= 0x1
+    index += 1
+    zeros = trailing_zeros(chunk % Int)
+    return index, (chunk >> (zeros % T), index + zeros)
+end
+
+function find_particle(a::FermiFS, i)
+    for k in occupied_orbitals(a)
+        i -= 1
+        i == 0 && return k
+    end
+    return 0
+end
+
+@inline function m_onr(a::FermiFS{<:Any,M}) where {M}
+    result = zero(MVector{M,Int32})
+    j = num_chunks(a.bs)
+    @inbounds for orbital in occupied_orbitals(a)
+        result[orbital] = 1
+    end
+    return result
+end
+
+function is_occupied(a::FermiFS{<:Any,M,S}, orbital) where {M,N,T,S<:BitString{<:Any,N,T}}
```

Comment:
Did you mean to do this:
```suggestion
function is_occupied(a::FermiFS{<:Any,M,S}, orbital) where {M,T,S<:BitString{<:Any,1,T}}
```
?
The next question is: Does it generate faster code than the generic method below?

Reply:
It should. `fldmod1` is surprisingly slow.